dbms/src/Storages/StorageLog.cpp

```diff
@@ -117,7 +111,7 @@ class LogBlockOutputStream final : public IBlockOutputStream
     explicit LogBlockOutputStream(StorageLog & storage_)
         : storage(storage_),
         lock(storage.rwlock),
-        marks_stream(storage.marks_file.path(), 4096, O_APPEND | O_CREAT | O_WRONLY)
+        marks_stream(fullPath(storage.disk, storage.marks_file_path), 4096, O_APPEND | O_CREAT | O_WRONLY)
```

Comment:
It's not integrated.

Reply:
All right. Only StorageTinyLog is fully integrated. It's left a few places in StorageLog and StorageStripeLog that require additional changes in buffer classes. I'd prefer to cover it in a separate PR.